### ISSUE TITLE
Add constant type inference for regex literals

### ISSUE THREAD

**Issue by lgo** [sorbet/sorbet#1421]
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0A%0Aextend%20T%3A%3ASig%0A%0Amodule%20Foo%0A%20%20RegexLiteral%20%3D%20%25r%7Bhello%7D%0A%20%20AnotherRegexLiteral%20%3D%20%2Fhello%2F%0Aend%0A%0A%23%20Both%20constants%20are%20T.untyped%2C%20but%20they%20are%20possible%20to%20infer%20as%20Regexp.%0AT.reveal_type(Foo%3A%3ARegexLiteral)%0AT.reveal_type(Foo%3A%3AAnotherRegexLiteral)%0A%0A%23%20For%20regular%20lvar%20assignment%2C%20the%20type%20is%20inferred.%0Aregex_literal%20%3D%20%25r%7Bhello%7D%0Aanother_regex_literal%20%3D%20%2Fhello%2F%0AT.reveal_type(regex_literal)%0AT.reveal_type(another_regex_literal)%0A)

```ruby
# typed: true

extend T::Sig

module Foo
  RegexLiteral = %r{hello}
  AnotherRegexLiteral = /hello/
end

# Both constants are T.untyped, but they are possible to infer as Regexp.
T.reveal_type(Foo::RegexLiteral)
T.reveal_type(Foo::AnotherRegexLiteral)

# For regular lvar assignment, the type is inferred.
regex_literal = %r{hello}
another_regex_literal = /hello/
T.reveal_type(regex_literal)
T.reveal_type(another_regex_literal)

```

#### Observed output

The constant types are `T.untyped`

#### Expected behavior

The constant types are inferred as `Regexp`

**Comment by jez** [comment]
I think we're unlikely to implement this as suggested, i.e., for Regexp literals. A feature which we might add some day is to re-use the logic we have for suggesting sigs on methods without sigs to also suggest sigs for constants.

**Comment by jez** [comment]
Fixed in #1692
